### DISCUSSION TITLE
fatal: 'git@github.com/Homebrew/brew' does not appear to be a git repository

### DISCUSSION THREAD

**Discussion by JP-93** [Homebrew/discussions#4028] [Getting started]
Hello guys!
When I try to install homebrew by command line, I get the following error:
==> Checking for `sudo` access (which may request your password)...
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/mkdir -p /usr/local/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R pedrobsouza:admin /usr/local/Homebrew
==> Downloading and installing Homebrew...
fatal: 'git@github.com/Homebrew/brew' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Failed during: git fetch --force origin

I tried to apply the solution from unset HOMEBREW_BREW_GIT_REMOTE HOMEBREW_CORE_GIT_REMOTE, but without success
Mac version Ventura 13.0.1

**Comment by baronnoraz** [comment]
I tried to manually clone the repo `git clone git@github.com/Homebrew/brew.git` and that failed.  However replacing the first forward slash with a colon works, `git clone git@github.com:Homebrew/brew.git`

I exported the correct repository URLs and was able to perform a `brew update` again.
```
export HOMEBREW_BREW_GIT_REMOTE=git@github.com:Homebrew/brew
export HOMEBREW_CORE_GIT_REMOTE=git@github.com:Homebrew/homebrew-core
```

Of course now I get...
```
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
fatal: 'git@github.com/Homebrew/homebrew-cask' does not appear to be a git repository
fatal: Could not read from remote repository.
```
I don't know how to fix - fix it, but maybe this helps.

**Reply by gromgit** [reply]
@JP-93 @baronnoraz What's the output of `git config -l | grep -i insteadof`?

**Reply by baronnoraz** [reply]
```
❯ git config -l | grep -i insteadof
url.git@github.com.insteadof=https://github.com
```

**Reply by JP-93** [reply]
url.git@github.com:mycompany.insteadof=https://github.com/mycompany
url.git@github.com:mycompany.insteadof=https://github.com/mycompany
url.git@github.com/.insteadof=https://github.com/

**Reply by gromgit** [reply]
>```
>url.git@github.com.insteadof=https://github.com
>```
@baronnoraz That literally replaces `https://github.com` with `git@github.com`.
>```
>url.git@github.com.insteadof=https://github.com/
>```
@JP-93 That literally replaces `https://github.com/` with `git@github.com/`.

In both cases `https://github.com/Homebrew/...` is transformed into `git@github.com/Homebrew/...`, which parses as a _directory pathname_ rather than an HTTPS or SSH URL.

Check your `~/.gitconfig` for a stanza that looks like this:
```
[url "git@github.com"]
        insteadOf = https://github.com
```
and replace it with _exactly_ this:
```
[url "git@github.com:"]
        insteadOf = https://github.com/
```
Then run `git config -l | grep -i insteadof` and confirm that it returns _exactly_ this:
```sh
$ git config -l | grep -i insteadof`
url.git@github.com:.insteadof=https://github.com/
```

**Reply by baronnoraz** [reply]
@gromgit that did the trick.  Thank you!!

**Reply by JP-93** [reply]
@baronnoraz  that did the trick. Thank you!!
